libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
+    count--;
+  }
+  if (ret) {
+    dst[ret-1] = '\0';
+  }
+  return -E2BIG;
```

Comment:
imho we should make r_str_ncpy remove that IS_PRINTABLE thing, and filter if needed in the following places:
```
$ git grep r_str_ncpy ..
../bin/format/mach0/mach0.c:                    r_str_ncpy (sectname, seg->segname, sizeof (sectname) - 1);
../bin/format/mach0/mach0.c:            r_str_ncpy (sectname, bin->sects[i].sectname, sizeof (sectname) - 1);
../core/disasm.c:                                       r_str_ncpy (s, (const char *)b, sizeof (s));
```